src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -44,8 +44,10 @@ internal class AnalyzerExecutor
         /// <param name="isCompilerAnalyzer">Delegate to determine if the given analyzer is compiler analyzer. 
         /// We need to special case the compiler analyzer at few places for performance reasons.</param>
         /// <param name="analyzerManager">Analyzer manager to fetch supported diagnostics.</param>
-        /// <param name="singleThreadedAnalyzerToGateMap">Map from non-thread safe analyzers to unique gate objects. 
-        /// All analyzer callbacks for these analyzers will be guarded with a lock on the gate.
+        /// <param name="getAnalyzerGate">
+        /// Delegate to fetch the gate object to guard all callbacks into the analyzer.
+        /// It should return a unique gate object for the given analyzer instance for non-concurrent analyzers, and null otherwise.
```

Comment:
Agreed, but unless we actually attempt to execute any dependent/related actions in parallel (our current drivers don't), a gate object is not required to guard just against concurrent callbacks for related actions.